### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by ermshiperete**
For this PR it's easiest to review the individual commits.

It contains several refactorings and cleanup and removes a warning on Linux.
- refactor alignment of parameters
- replace tabs with spaces
- always output function name in debug messages
- comment unused methods. This solves #7297.

~- remove a warning on Linux~

~@rc-swag Please take a look at 9de3000fa9911bd426561b99ef41650a256678aa since that might affect Windows. I don't think the `KMN_API` is needed, but I might be wrong.~

I'm fine if we don't want to merge this at this point so close to beta, but since I had figured out what methods get called and are needed while debugging #7084 I didn't want to throw away what I had.

@keymanapp-test-bot skip

### CODE REVIEW

File: core/include/keyman/keyboardprocessor.h [hunk 1]
```diff
@@ -127,7 +127,7 @@ typedef struct km_kbp_option_item  km_kbp_option_item;
 // Callback function used to to access Input Method eXtension library functions
 // from Keyman Core
 //
-typedef KMN_API uint8_t (*km_kbp_keyboard_imx_platform)(km_kbp_state*, uint32_t, void*);
+typedef uint8_t (*km_kbp_keyboard_imx_platform)(km_kbp_state*, uint32_t, void*);
```
Comment (mcdurdin) [inline core/include/keyman/keyboardprocessor.h:130]: `KMN_API` is a calling convention macro, so this will definitely break on Windows.

Comment (ermshiperete) [reply core/include/keyman/keyboardprocessor.h]: ok, I'll remove this change from this PR and create an issue for the warning.

File: linux/ibus-keyman/src/engine.c [hunk 2]
```diff
@@ -517,11 +515,11 @@ ibus_keyman_engine_destroy (IBusKeymanEngine *keyman)
 }
 
 static void
-ibus_keyman_engine_commit_string (IBusKeymanEngine *keyman,
+commit_string (IBusKeymanEngine *keyman,
                                 const gchar    *string)
```
Comment (mcdurdin) [inline linux/ibus-keyman/src/engine.c:519]:
```suggestion
               const gchar    *string)
```

File: linux/ibus-keyman/src/engine.c [hunk 3]
```diff
@@ -974,17 +971,17 @@ ibus_keyman_engine_focus_out (IBusEngine *engine)
...
 }
 
 static void
 ibus_keyman_engine_reset (IBusEngine *engine)
 {
-    g_message("ibus_keyman_engine_reset");
+    g_message("%s", __FUNCTION__);
     parent_class->reset (engine);
-    ibus_keyman_engine_focus_in (engine);
+    reset_context(engine);
```
Comment (mcdurdin) [inline linux/ibus-keyman/src/engine.c:984]: What is the purpose of this change?

**Review by mcdurdin — CHANGES_REQUESTED** [review]
I'm a little uncomfortable with all these changes being in the one PR. I'd prefer to have formatting changes separated out into their own PR, the debugging changes in a second PR, and then each of the functional changes clearly identified and potentially separated out as well so we can verify and test them in isolation.

Right now, this PR has the title 'Refactorings', which suggests no functional change, but actually this PR has several functional changes. (Also it should at least be 'chore(linux): refactorings' or something like that!)

**Comment by jahorton** [comment]
> Right now, this PR has the title 'Refactorings', which suggests no functional change, but actually this PR has several functional changes. (Also it should at least be 'chore(linux): refactorings' or something like that!)

Keep in mind, our automated `history.md` management actually uses your PR title for history notes.  "Refactorings" is far less informative than something more explicit about the changes involved.

File: linux/ibus-keyman/src/engine.c (hunk 3 above)
Comment (ermshiperete) [reply linux/ibus-keyman/src/engine.c:984]: Both `ibus_keyman_engine_reset` and `ibus_keyman_engine_focus_in` are methods that get called by ibus. Having one call the other makes it much harder to see from a log file what methods ibus actually calls.

**Comment by ermshiperete** [comment]
> I'm a little uncomfortable with all these changes being in the one PR. I'd prefer to have formatting changes separated out into their own PR, the debugging changes in a second PR, and then each of the functional changes clearly identified and potentially separated out as well so we can verify and test them in isolation.
> 
> Right now, this PR has the title 'Refactorings', which suggests no functional change, but actually this PR has several functional changes. (Also it should at least be 'chore(linux): refactorings' or something like that!)

You're right. Not being able to give a concise title should have hinted me that there are two many unrelated changes...

Closing this PR and replacing it with several new ones.

File: linux/ibus-keyman/src/engine.c (hunk 3 above)
Comment (mcdurdin) [reply linux/ibus-keyman/src/engine.c:984]: Great, that makes sense, thanks!

**Comment by mcdurdin** [comment]
> You're right. Not being able to give a concise title should have hinted me that there are two many unrelated changes...
> 
> Closing this PR and replacing it with several new ones: #7499, #7500, #7501, #7502, #7503

Awesome thank you! All the replacement PRs look great!
